util/argo/normalizers/diff_normalizer.go

```diff
@@ -179,6 +180,12 @@ func (n *ignoreNormalizer) Normalize(un *unstructured.Unstructured) error {
 	for _, patch := range matched {
 		patchedDocData, err := patch.Apply(docData)
 		if err != nil {
+			if strings.Contains(err.Error(), "Unable to remove nonexistent key") {
+				continue
+			}
+			if strings.Contains(err.Error(), "remove operation does not apply: doc is missing path") {
+				continue
+			}
```

Comment:
It's a bit unfortunate we have to detect these this way.

Would you mind adding unit tests for this to avoid regressions if the library changes its strings?